### ISSUE TITLE
some question about stage3 klloss

### ISSUE THREAD

**Issue by bollossom** [hustvl/mmMamba#4]
I found that the kl loss of stage3 is as high as 8000. Is this normal?

![Image](https://github.com/user-attachments/assets/322c1817-c048-492e-918e-af479d688f8e)

**Comment by bollossom** [comment]
moreover, the following is my MSEloss in stage 1 and stage 2

![Image](https://github.com/user-attachments/assets/f13903e4-bf4e-4d48-ada1-0c5b47f3d208)

![Image](https://github.com/user-attachments/assets/a9bcafd2-01bc-456b-91a8-6625d98bba62)

**Comment by Hongyuan-Tao** [comment]
Hi! Thank you for your feedback! 
The loss decline in Stage 1 and Stage 2 seems fine. However, there is indeed a problem with Stage 3. It appears that it was not initialized properly. I will recheck the initialization program for Stage 3 later.

**Comment by Hongyuan-Tao** [comment]
Just wanted to check: In the training curves of stage1 and stage2 you sent me, is the picture on top for Stage 1 and the one below for Stage 2?

**Comment by Hongyuan-Tao** [comment]
It seems that each stage didn't initialize from the checkpoint of the previous stage. Could you check if you added the `--load_distill_checkpoint` option when starting the training program to ensure proper initialization?

**Comment by bollossom** [comment]
> Just wanted to check: In the training curves of stage1 and stage2 you sent me, is the picture on top for Stage 1 and the one below for Stage 2?

Thanks for your quick reply. Top is the stage 2. Bottom is the stage 1.

**Comment by bollossom** [comment]
> It seems that each stage didn't initialize from the checkpoint of the previous stage. Could you check if you added the `--load_distill_checkpoint` option when starting the training program to ensure proper initialization?

Thanks for your quick reply.  Here is my stage3 train.sh


```
accelerate launch \
  --config_file  "./configs/accelerate/default_config.yaml"  \
  --main_process_port  "12345"  \
    distill_mmMamba.py \
  --model_config distill_mmMamba \
  --distill_stage3_config distill_stage3_mmMamba \
  --checkpoint_dir ./checkpoints \
  --train_stage3  \
  --verbose \
  --seed 0 \
  --replicate 0   \
  --load_distill_checkpoint ./checkpoints/1_distill-se=0-re=0_distill_all_stage2_last.pt
```

**Comment by Hongyuan-Tao** [comment]
It seems fine. I'll rerun the code to test it. Please wait a moment.

**Comment by bollossom** [comment]
> It seems fine. I'll rerun the code to test it. Please wait a moment.

Thanks for your quick and detailed reply.

**Comment by Hongyuan-Tao** [comment]
Hi! 
    We have reviewed and tested the entire codebase, and also examined our previous training log files. The conclusion we have reached now is that there are no issues with your training at each stage, and the changes in loss are generally normal. 
    It is important to note that line 78 in the file` configs/experiment/distill_stage3_mmMamba.yaml`  specifies `softmax_attention` as `[0, 4, 8, 12, 16, 20, 24, 28]`. This configuration indicates that the model is set to hybrid mode. Later, I will switch it to linear mode by setting `softmax_attention` to an empty list (`[]`).  And we will update our code to fix some minor issues within it, but this will not affect your current training.
   We believe you can proceed with the training using the current settings and then conduct relevant tests on the model to examine its performance. If you encounter any issues later on, feel free to continue discussing with us.

**Comment by bollossom** [comment]
Thank you for your prompt reply. Could you please release the KLloss and startup command and checkpoint of stage 3?

**Comment by Hongyuan-Tao** [comment]
This is a screenshot of a portion of the log from one of our stage3 training sessions, for your reference.

![Image](https://github.com/user-attachments/assets/95f01263-0736-4d4f-bae4-e6190c1abaae)

**Comment by Hongyuan-Tao** [comment]
This is a screenshot of a portion of the log from one of our stage3 training sessions, for your reference.
![Image](https://github.com/user-attachments/assets/8461aa83-eee5-4a49-b357-465681bdc870)

**Comment by bollossom** [comment]
Thanks for you quick reply. I have no more questions. Have a nice day!!!
